src/connector.cpp

```diff
@@ -5193,19 +5025,92 @@ void HostConnector::hostSwitchPreset(const Current& prev)
                     if (defpropdata.value == defpropdata.meta.defpath)
                         continue;
 
-                    _host.patch_set(hbp.id, defpropdata.uri.c_str(), defpropdata.value.c_str());
-
-                    if (hbp.pair != kMaxHostInstances)
-                        _host.patch_set(hbp.pair, defpropdata.uri.c_str(), defpropdata.value.c_str());
+                    hostPatchSetBlockPair(hbp, defpropdata);
                 }
 
-                _host.params_flush(hbp.id, LV2_KXSTUDIO_PROPERTIES_RESET_FULL, params.size(), params.data());
-
-                if (hbp.pair != kMaxHostInstances)
-                    _host.params_flush(hbp.pair, LV2_KXSTUDIO_PROPERTIES_RESET_FULL, params.size(), params.data());
+                hostParamsFlushBlockPair(hbp, LV2_KXSTUDIO_PROPERTIES_RESET_FULL, params);
             }
         }
+        // ensure necessary dual mono blocks are added
+        hostEnsureStereoChain(prev.preset, 0, 0);
```

Comment:
calling this function with 0, 0 as args looks weird... we should probably then make those args optional with a default value